langfuse/decorators/langfuse_decorator.py

```diff
@@ -477,7 +485,7 @@ def get_current_langchain_handler(self):
 
         return observation.get_langchain_handler()
 
-    def get_current_trace_id(self):
+    def get_current_trace_id(self, log_warnings: bool = True):
```

Comment:
Wdyt about always logging a warning here? I think this would be unexpected for the user to not get an id here.

Reply:
I see the reason for the change below. In case of the openai integration, we do not want to log error messages. What do you think about adding the logic of these functions to the openai integration as well? 
Reasoning:
- these functions here are fairly shallow
- If we add log_warnings to the public interface, we need to explain them and cannot just remove them. Hence i would only expose what is necessary.